sdk/java-sdk-spring/src/main/scala/kalix/javasdk/client/ComponentCall.scala

```diff
@@ -57,15 +57,23 @@ final class ComponentCall[A1, R](
 
 object ComponentCall {
 
-  def noParams[R](kalixClient: KalixClient, lambda: scala.Any, ids: util.List[String]): DeferredCall[Any, R] = {
-    invoke(Seq.empty, kalixClient, MethodRefResolver.resolveMethodRef(lambda), ids.asScala.toList)
+  def noParams[R](
+      kalixClient: KalixClient,
+      lambda: scala.Any,
+      ids: util.List[String],
+      callMetadata: Optional[Metadata]): DeferredCall[Any, R] = {
+    invoke(Seq.empty, kalixClient, MethodRefResolver.resolveMethodRef(lambda), ids.asScala.toList, callMetadata)
   }
 
-  def noParams[R](kalixClient: KalixClient, method: Method, ids: util.List[String]): DeferredCall[Any, R] = {
-    invoke(Seq.empty, kalixClient, method, ids.asScala.toList)
+  def noParams[R](
+      kalixClient: KalixClient,
+      method: Method,
+      ids: util.List[String],
+      callMetadata: Optional[Metadata]): DeferredCall[Any, R] = {
+    invoke(Seq.empty, kalixClient, method, ids.asScala.toList, callMetadata)
```

Comment:
Actually, Those method should be private.

Reply:
We can only make them private with `private[client]` and that is ignored by java, so not really useful. I will leave it as such.

The right thing to do is to move it to an `impl` package.